internal/webserver/handler-api.go

```diff
@@ -458,6 +465,103 @@ func (h *handler) apiUpdateBookmark(w http.ResponseWriter, r *http.Request, ps h
 	checkError(err)
 }
 
+// apiDownloadEbook is handler for PUT /api/ebook
+func (h *handler) apiDownloadEbook(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	ctx := r.Context()
+
+	// Make sure session still valid
+	err := h.validateSession(r)
+	checkError(err)
+
+	// Decode request
+	request := struct {
+		IDs []int `json:"ids"`
+	}{}
+	err = json.NewDecoder(r.Body).Decode(&request)
+	checkError(err)
+
+	// Get existing bookmark from database
+	filter := database.GetBookmarksOptions{
+		IDs:         request.IDs,
+		WithContent: true,
+	}
+
+	bookmarks, err := h.DB.GetBookmarks(ctx, filter)
+	checkError(err)
+	if len(bookmarks) == 0 {
+		panic(errors.Wrap(err, "no bookmark with matching ids"))
+	}
+
+	// Fetch data from internet
+	mx := sync.RWMutex{}
+	wg := sync.WaitGroup{}
+	chDone := make(chan struct{})
+	chProblem := make(chan int, 10)
+	semaphore := make(chan struct{}, 10)
```

Comment:
I'm assuming this semaphore is to limit concurrent download to this channel size, right?